### DISCUSSION TITLE
Inquiry About Offline Usability and API Key/Email Input Requirements

### DISCUSSION THREAD

**Discussion by sungreong** [vanna-ai/vanna#215] [Q&A] [answered]
Hello Vanna AI Team,

I am currently exploring the capabilities of the Vanna AI and am interested in implementing it in a project. However, I have a couple of questions that I couldn't find answers to in the documentation:

1. **Offline Usability**: Is it possible to use Vanna AI in an environment without an internet connection? If so, are there any limitations or specific configurations needed to operate it offline?

2. **API Key/Email Requirements**: The documentation often mentions entering an API key or an email. Can you clarify if this is mandatory for all operations, and what the key/email is used for? Additionally, in an offline scenario, how would

**Comment by zainhoda** [comment]
Full offline usability is pending the ollama support -- I anticipate that it should be ready some reasonably soon.
https://github.com/vanna-ai/vanna/issues/167

The API key / email is only necessary if you're choosing a configuration that uses any of Vanna's hosted services (OpenAI via Vanna or Vanna-Hosted Vector Database).

If you have your own OpenAI API key, you can choose to use that and then use the local vector database option (ChromaDB).

**Reply by zainhoda** [reply, marked as answer]
Ollama support is now present so you can choose a fully offline deployment without the need for an API-based LLM

https://vanna.ai/docs/postgres-ollama-chromadb/
